openmdao/visualization/record_viewer.py

```diff
@@ -0,0 +1,401 @@
+        if self.io_select_x.value == "Case Iterations":
+            self._case_iter_x = True
+
+        if len(self.case_options) != 1 and (self._case_iter_x or self._case_iter_y):
+            if len(self.case_select.value) == 1:
+                current_val = int(self.case_select.value[0])
+                self.case_select.value.append(str(current_val + 1))
+        elif self._case_iter_x or self._case_iter_y:
+            self.warning_box.text = ("NOTE: Only one case found. Unable to view case iterations vs "
+                                     "variable")
+
+        for i in self.case_select.value:
+            self.case = self.cr.get_case(self.case_options[int(i)][1])
+
+            for key, val in self.io_options_x.items():
+                if self.io_select_y.value in val:
+                    y_io = getattr(self.case, key)
+
+                if self.io_select_x.value in val:
+                    x_io = getattr(self.case, key)
+
+            if (num_points_y and num_points_x) or (self._case_iter_x and self._case_iter_y):
+                x_variable = np.zeros(1)
+                y_variable = np.zeros(1)
+            elif (num_points_x and self._case_iter_y) or (num_points_y and self._case_iter_x):
+                x_variable = np.zeros(1)
+                y_variable = np.zeros(1)
+                self.warning_box.text = "NOTE: Cannot compare Number of Points to Case Iterations"
+            elif num_points_y or self._case_iter_y:
+                x_variable = x_io[self.io_select_x.value].flatten()
+                y_variable = np.arange(len(x_variable))
+            elif num_points_x or self._case_iter_x:
+                y_variable = y_io[self.io_select_y.value].flatten()
+                x_variable = np.arange(len(y_variable))
+            else:
+                x_variable = x_io[self.io_select_x.value].flatten()
+                y_variable = y_io[self.io_select_y.value].flatten()
+
+            if not isinstance(new_data['x_vals'], np.ndarray):
+                new_data['x_vals'] = np.empty((0, len(x_variable)), float)
+                new_data['y_vals'] = np.empty((0, len(y_variable)), float)
+
+            new_data['x_vals'] = np.vstack((new_data['x_vals'], x_variable))
+            new_data['y_vals'] = np.vstack((new_data['y_vals'], y_variable))
+
+        x_len = new_data['x_vals'].shape[1]
+        y_len = new_data['y_vals'].shape[1]
+        new_data['color'] = self._line_color_list(new_data['x_vals'])
+        new_data['cases'] = [self.case_options[int(case)][1] for case in self.case_select.value]
+        case_len = len(new_data['cases'])
+
+        if new_data['x_vals'].shape[1] > 1:
+            if (new_data['x_vals'].shape[0], new_data['y_vals'].shape[0]) == (1, 1) and \
+               (set(new_data['x_vals'][0]), set(new_data['y_vals'][0])) == ({0.}, {0.}):
+                self.warning_box.text = """NOTE: Both X and Y values contain zeros for values,
+                                        unable to plot"""
```

Comment:
looks like you will get a bunch of spaces between `values,` and `unable`... is that intentional?